### DISCUSSION TITLE
How to plot multiple points at single time step?

### DISCUSSION THREAD

**Discussion by dc217863** [facontidavide/PlotJuggler#565] [Q&A] [answered]
Hi,
I am using ROS and have the following use case:

Points.msg:
std_msgs/Header header
float64[] x
float64[] y

I have a topic containing multiple x, y points in a single message. I would like to plot all the x,y points at a particular time step. At the next time step I have new x,y points I would like to plot.

Is this possible with PlotJuggler? I understand that I have to create a plugin to parse the data from the messages as I want it. However, I need some guidance to understand whether it is possible to plot multiple points at a single timestep.

Thanks!

**Comment by dc217863** [comment]
I had a quick look at the code and am currently working with the following idea:
create a plugin to parse the information from the msg and add it to the series (similar to ros1_parsers::PoseMsgParser::parseMessageImpl()). This would lead to having multiple points at the same timestamp. This way no new data structure would be required.

However, the question remains, do points get automatically plotted if they are the same timestamp? If not, do you have a suggestion of how to go about this?

**Reply by facontidavide** [reply]
Yes, writing your own parser is certainly the most effective idea, but SDince version 3.4.x there is a new powerful mechanism to deal with this kind of data.

See https://slides.com/davidefaconti/plotjuggler-reactive-scripts/fullscreen

**Comment by Wayne-DWA** [comment]
@facontidavide Hi, I followed your slides and this [answer](https://answers.ros.org/question/321890/visualize-or-plot-float32multiarray-data-field/) but I always got this error when I clicked "save" the function. Is there a bug or something I did wrong? Thanks.
![plotjugger](https://user-images.githubusercontent.com/90710539/183053271-6d4f69ea-9937-4ed6-a730-8a049901188e.png)

**Comment by facontidavide** [comment, marked as answer]
**MutableScatterXY** was renamed **ScatterXY**.

Tutorial updated

**Reply by Wayne-DWA** [reply]
thanks, it works
